src/Platforms/AbstractPlatform.php

```diff
@@ -4277,7 +4313,38 @@ public function supportsColumnCollation()
      */
     public function getDateTimeFormatString()
     {
-        return 'Y-m-d H:i:s';
+        return $this->getDateFormatString() . ' ' . $this->getTimeFormatString();
+    }
+
+    /**
+     * Gets the format string, as accepted by the date() function, that describes
+     * the format of a stored datetime value of this platform excluding any millisecond
+     * or microsecond specifier.
+     */
+    public function getFallbackFormatFromTimeFormatString(string $format): string
+    {
+        return $this->removeHighPrecisionTimePHP($format);
+    }
+
+    /**
+     * Whether the platform offers a fallback time format in the event that the first conversion to a
+     * php datetime object fails.
+     *
+     * Conversion of a time without fractional seconds to a php format which includes fractional seconds
+     * (either .v or .u) will fail.
+     */
+    public function hasFallbackTimeFormat(): bool
+    {
+        return $this->getDateTimeFormatString() !==
+               $this->getFallbackFormatFromTimeFormatString($this->getDateTimeFormatString());
+    }
+
+    /**
+     * Removes millisecond and microsecond specifiers (.v and .u) from datetime format if present.
+     */
+    private function removeHighPrecisionTimePHP(string $format): string
+    {
+        return preg_replace('/s(\.[uv])?([^\.])?/', 's$2', $format, 1) ?? $format;
```

Comment:
This looks like another hack.

Reply:
Yes. I've updated the patch to define `getFallbackDateTimeFormatString()`, `getFallbackDateTimeTzFormatString()` and `getFallbackTimeFormatString()` for each platform and use those methods instead of calling the above method on the existing format strings.